src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1466,6 +1468,16 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def getBucketUsage(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
+        DBIO.from(gcsDAO.getBucketUsage(workspaceContext.workspace.bucketName)) map { usage =>
+          RequestComplete(BucketUsageResponse(usage))
```

Comment:
we have learned that it is a bad idea to include external service calls in a transaction. Try
```
def getBucketUsage(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
  dataSource.inTransaction { dataAccess =>
    withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
      workspaceContext.workspace.bucketName
    }
  } flatMap { bucketName =>
    gcsDAO.getBucketUsage(bucketName))
  } map { usage =>
    RequestComplete(BucketUsageResponse(usage))
  }
}
```
or better yet
```
def getBucketUsage(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
  for {
    bucketName <- dataSource.inTransaction { dataAccess =>
      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
        workspaceContext.workspace.bucketName
      }
    }

    usage <- gcsDAO.getBucketUsage(bucketName))

  } yield {
    RequestComplete(BucketUsageResponse(usage))
  }
}
```